java/team25core/ObjectDetectionTask.java

```diff
@@ -0,0 +1,250 @@
+        tfodParameters.minResultConfidence = 0.8f; //the example in the Ultimate Goal Tensor flow example defaults to a MinimumConfidence of 0.8f
+        tfodParameters.isModelTensorFlow2 = true;
+        tfodParameters.inputSize = 320;
+
+        //concept tensor flow object detection had minimum confidence
+        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia);
+        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_OBJECT1);
+    }
+
+    public void init(Telemetry telemetry, HardwareMap hardwareMap)
+    {
+        initVuforia(hardwareMap);
+
+        // if (ClassFactory.getInstance().canCreateTFObjectDetector()) {
+            initTfod(hardwareMap);
+//        } else {
+//            telemetry.addData("Sorry!", "This device is not compatible with TFOD");
+//        }
+    }
+
+    public void rateLimit(int ms)
+    {
+        this.rateLimitMs = ms;
+    }
+
+    public void setDetectionKind(DetectionKind detectionKind)
+    {
+        this.detectionKind = detectionKind;
+    }
+    //this will start tfod activation and start detecting
+    @Override
+    public void start()
+    {
+        tfod.activate();
+
+        if (rateLimitMs != 0) {
+            timer = new ElapsedTime(ElapsedTime.Resolution.MILLISECONDS);
+        }
+    }
+
+    @Override
+    public void stop()
+    {
+        tfod.deactivate();
+        robot.removeTask(this);
+    }
+
+    public static ObjectKind isObject(Recognition object)
+    {
+        if (object.getLabel().equals(LABEL_OBJECT1)) {
+            return ObjectKind.OBJECT1_KIND;
+        } else {
+            return ObjectKind.UNKNOWN_KIND;
+        }
+    }
+    //if recognize anything will add to que
+    protected void processEverything(List<Recognition> objects)
+    {
+        if (objects.size() > 0) {
+            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objects));
+        }
+    }
+    //only adds objects which will make event and add to que
+    protected void processObject1(List<Recognition> objects)
+    {
+        List<Recognition> objectList = new ArrayList<>();
+        for (Recognition object : objects) {
+            if (isObject(object) == ObjectKind.OBJECT1_KIND){
+                objectList.add(object);
+            }
+        }
+
+        if (!objectList.isEmpty()) {
+            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objectList));
+        }
+    }
+
+//    protected void processObject2(List<Recognition> objects)
+//    {
+//        List<Recognition> objectList = new ArrayList<>();
+//        for (Recognition object : objects) {
+//            if (isObject(object) == ObjectKind.OBJECT2_KIND) {
+//                objectList.add(object);
+//            }
+//        }
+//
+//        if (!objectList.isEmpty()) {
+//            robot.queueEvent(new ObjectDetectionEvent(this, EventKind.OBJECTS_DETECTED, objectList));
+//        }
+//    }
```

Comment:
Here you could consolidate processObject1 and processObject2 into one method called processObject that takes as a parameter the ObjectKind.

e.g.

```
protected void processObject(List<Recognition> objects, ObjectKind kind) { .... }
```
And then you call it and pass the ObjectKind you want to use in the if expression on line 189.